packages/block/src/from-rpc.ts

```diff
@@ -1,35 +1,30 @@
 import { TransactionFactory } from '@ethereumjs/tx'
-import { setLengthLeft, toBuffer } from '@ethereumjs/util'
+import { TypeOutput, setLengthLeft, toBuffer, toType } from '@ethereumjs/util'
 
 import { blockHeaderFromRpc } from './header-from-rpc'
-import { numberToHex } from './helpers'
 
 import { Block } from './index'
 
 import type { BlockOptions, JsonRpcBlock } from './index'
 import type { TxData, TypedTransaction } from '@ethereumjs/tx'
 
-function normalizeTxParams(_txParams: any) {
+export function normalizeTxParams(_txParams: any) {
   const txParams = Object.assign({}, _txParams)
 
-  txParams.gasLimit = txParams.gasLimit === undefined ? txParams.gas : txParams.gasLimit
+  txParams.gasLimit = toType(txParams.gasLimit ?? txParams.gas, TypeOutput.BigInt)
   txParams.data = txParams.data === undefined ? txParams.input : txParams.data
 
   // check and convert gasPrice and value params
-  txParams.gasPrice = numberToHex(txParams.gasPrice)
-  txParams.value = numberToHex(txParams.value)
+  txParams.gasPrice = txParams.gasPrice !== undefined ? BigInt(txParams.gasPrice) : undefined
+  txParams.value = txParams.value !== undefined ? BigInt(txParams.value) : undefined
 
   // strict byte length checking
   txParams.to =
     txParams.to !== null && txParams.to !== undefined
       ? setLengthLeft(toBuffer(txParams.to), 20)
       : null
 
-  // v as raw signature value {0,1}
-  // v is the recovery bit and can be either {0,1} or {27,28}.
-  // https://ethereum.stackexchange.com/questions/40679/why-the-value-of-v-is-always-either-27-11011-or-28-11100
-  const v: number = txParams.v
-  txParams.v = v < 27 ? v + 27 : v
+  txParams.v = toType(txParams.v, TypeOutput.BigInt)!
```

Comment:
```suggestion
  txParams.v = toType(txParams.v, TypeOutput.BigInt)
```
I don't think the ! is necessary